stock-service/src/main/java/com/ecommercepractice/stockservice/controller/StockController.java

```diff
@@ -0,0 +1,106 @@
+        return new ResponseEntity(stockModelAssembler.toModel(stockService.appendToInventory(productStock)), HttpStatus.CREATED);
+    }
+
+    @ApiOperation(value ="FILTER", notes = "Fetch products by filtering them by price, stockQuantity and soldQuiantity" +
+            "Also, you can specify whether you want to list not onlyProducts also")
+    @GetMapping()
+    public ResponseEntity<List<EntityModel<Stock>>>fetchByFiltering(@RequestParam(required = false) Float price,
+                                                        @RequestParam(required = false) Long stockQuantity,
+                                                        @RequestParam(required = false) Long soldQuantity,
+                                                        @RequestParam(required = false, defaultValue = "true") Boolean onlyActive) {
+        log.info(String.format("STOCK | FETCH | STOCK NAME -> { %s } PRICE -> { %s }  ",
+                price == null ? "NO_PRICE" : price.toString(),
+                stockQuantity == null ? "NO_stockQuantity" : stockQuantity.toString(),
+                soldQuantity == null ? "NO_soldQuantity" : soldQuantity.toString()));
+
+        List<EntityModel<Stock>> stockList = stockService.fetchByFiltering(price, stockQuantity, soldQuantity, onlyActive)
+                .stream()
+                .map(stockModelAssembler::toModel)
+                .collect(Collectors.toList());
+
+        return new ResponseEntity(stockList, HttpStatus.OK);
+    }
+
+    @ApiOperation(value ="FETCH BY ID", notes = "Get a product on stock by their id.")
+    @GetMapping("/{stockId}")
+    public ResponseEntity<EntityModel<Stock>> fetchByStockId(@PathVariable(required = true) Long stockId) {
+        log.info(String.format("STOCK | FETCH | PAYLOAD { StockId -> %s } ", stockId));
+        return new ResponseEntity(stockModelAssembler.toModel(stockService.fetchByStockId(stockId)), HttpStatus.OK);
+    }
+
+    @ApiOperation(value ="UPDATE BY ID", notes = "Update stockQuantity and/or salePrice on stock by their id.")
+    @PutMapping("/{stockId}")
+    public ResponseEntity<EntityModel<Stock>> updateByStockId(@PathVariable(required = true) Long stockId,
+                                                 @RequestParam(required = false) Long stockQuantity,
+                                                 @RequestParam(required = false) Float salePrice
+    ) {
+        log.info(String.format("STOCK | UPDATE | PAYLOAD { StockId -> %s , Stock -> %s } ", stockId,
+                stockQuantity == null ? "NO_QUANTITY" : stockQuantity,
+                salePrice == null ? "NO_SALE_PRICE" : salePrice
+        ));
+        return new ResponseEntity(stockModelAssembler.toModel(stockService.updateByStockId(stockId, stockQuantity, salePrice)), HttpStatus.OK);
+    }
+    @ApiOperation(value ="DELETE BY ID", notes = "DELETE a product on stock by their id.")
+    @DeleteMapping("/{stockId}")
```

Comment:
please, try to set a resource more clear for example /stock/products/{product-id}

Reply:
done